packages/core/schematics/ng-generate/standalone-migration/prune-modules.ts

```diff
@@ -0,0 +1,365 @@
+
+  return {pendingChanges: tracker.recordChanges(), filesToRemove};
+}
+
+/**
+ * Collects all the nodes that a module needs to be removed from.
+ * @param ngModule Module being removed.
+ * @param removalLocations
+ * @param languageService
+ * @param program
+ */
+function collectRemovalLocations(
+    ngModule: ts.ClassDeclaration, removalLocations: RemovalLocations,
+    languageService: ts.LanguageService, program: NgtscProgram) {
+  const refsByFile = extractReferences(ngModule, languageService);
+  const tsProgram = program.getTsProgram();
+  const nodes = new Set<ts.Node>();
+
+  for (const [fileName, refs] of refsByFile) {
+    const sourceFile = tsProgram.getSourceFile(fileName);
+
+    if (sourceFile) {
+      offsetsToNodes(getNodeLookup(sourceFile), refs, nodes);
+    }
+  }
+
+  for (const node of nodes) {
+    const closestArray = closestNode(node, ts.isArrayLiteralExpression);
+    if (closestArray) {
+      removalLocations.arrays.track(closestArray, node);
+      continue;
+    }
+
+    const closestImport = closestNode(node, ts.isNamedImports);
+    if (closestImport) {
+      removalLocations.imports.track(closestImport, node);
+      continue;
+    }
+
+    const closestExport = closestNode(node, ts.isNamedExports);
+    if (closestExport) {
+      removalLocations.exports.track(closestExport, node);
+      continue;
+    }
+
+    removalLocations.unknown.add(node);
+  }
+}
+
+/**
+ * Removes all tracked array references.
+ * @param locations Locations from which to remove the references.
+ * @param tracker Tracker in which to register the changes.
+ */
+function removeArrayReferences(
+    locations: UniqueItemTracker<ts.ArrayLiteralExpression, ts.Node>,
+    tracker: ChangeTracker): void {
+  for (const [array, toRemove] of locations.getEntries()) {
+    const newElements = filterRemovedElements(array.elements, toRemove);
+    tracker.replaceNode(array, ts.factory.updateArrayLiteralExpression(array, newElements));
+  }
+}
+
+/**
+ * Removes all tracked import references.
+ * @param locations Locations from which to remove the references.
+ * @param tracker Tracker in which to register the changes.
+ */
+function removeImportReferences(
+    locations: UniqueItemTracker<ts.NamedImports, ts.Node>, tracker: ChangeTracker) {
+  for (const [namedImports, toRemove] of locations.getEntries()) {
+    const newElements = filterRemovedElements(namedImports.elements, toRemove);
+
+    // If no imports are left, we can try to drop the entire import.
+    if (newElements.length === 0) {
+      const importClause = closestNode(namedImports, ts.isImportClause);
+
+      // If the import clause has a name we can only drop then named imports.
+      // e.g. `import Foo, {ModuleToRemove} from './foo';` becomes `import Foo from './foo';`.
+      if (importClause && importClause.name) {
+        tracker.replaceNode(
+            importClause,
+            ts.factory.createImportClause(importClause.isTypeOnly, importClause.name, undefined));
```

Comment:
Nit: to preserve existing node info as much as possible, like formatting etc.

```suggestion
            ts.factory.updateImportClause(importClause, importClause.isTypeOnly, importClause.name, undefined));
```